### PR TITLE
TiKV config: Update some values and add new online configurations

### PR DESCRIPTION
**Pull request by en-jin19**
### What is changed, added or deleted? (Required)
1. Added the following new online configurations: `raftstore.raft-max-size-per-msg`, `raftstore.apply-max-batch-size`, `raftstore.store-max-batch-size`, `storage.scheduler-worker-pool-size`
2. Updated descriptions of the following online configurations: `raftstore.apply-pool-size`, `raftstore.store-pool-size`
3. Updated descriptions and values of the following TiKV configurations: `scheduler-worker-pool-size`, `raft-max-size-per-msg`, `apply-max-batch-size`, `apply-pool-size`, `store-max-batch-size`

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions](https://github.com/pingcap/docs/blob/master/CONTRIBUTING.md#guideline-for-choosing-the-affected-versions).

- [x] master (the latest development version)
- [ ] v6.0 (TiDB 6.0 versions)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)
- [ ] v5.0 (TiDB 5.0 versions)
- [ ] v4.0 (TiDB 4.0 versions)
- [ ] v3.1 (TiDB 3.1 versions)
- [ ] v3.0 (TiDB 3.0 versions)
- [ ] v2.1 (TiDB 2.1 versions)

### What is the related PR or file link(s)?

- This PR is translated from: [#8644](https://github.com/pingcap/docs-cn/pull/8644), [#8784](https://github.com/pingcap/docs-cn/pull/8784)
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

File: dynamic-config.md [hunk 1]
```diff
@@ -131,6 +131,7 @@ The following TiKV configuration items can be modified online:
 | `raftstore.raft-log-gc-threshold` | The soft limit on the maximum allowable number of residual Raft logs |
 | `raftstore.raft-log-gc-count-limit` | The hard limit on the allowable number of residual Raft logs |
 | `raftstore.raft-log-gc-size-limit` | The hard limit on the allowable size of residual Raft logs |
+| `raftstore.raft-max-size-per-msg` | The soft limit on the allowable size of a single message packet |
```
Comment (en-jin19) [inline dynamic-config.md:134]:
```suggestion
| `raftstore.raft-max-size-per-msg` | The soft limit on the size of a single message packet that is allowed to be generated |
```

**Review by glorv — COMMENTED** [review]
LGTM

**Review by glorv — APPROVED** [review]

File: tikv-configuration-file.md [hunk 2]
```diff
@@ -490,8 +490,9 @@ Configuration items related to Raftstore.
 
 + The soft limit on the size of a single message packet
 + Default value: `"1MB"`
-+ Minimum value: `0`
-+ Unit: MB
++ Minimum value: greater than `0`
```
Comment (TomShawn) [inline tikv-configuration-file.md:493]:
```suggestion
+ Minimum value: Greater than `0`
```

Comment (en-jin19) [reply tikv-configuration-file.md:493]: > 

How about using the lower cases as other lines (for example 501, 538)?

File: tikv-configuration-file.md [hunk 3]
```diff
@@ -727,25 +728,27 @@ Configuration items related to Raftstore.
...
-+ The maximum number of requests for data flushing in one batch
++ Raft state machines execute data write requests in batches by the BatchSystem. This configuration item specifies the maximum number of Raft state machines that can execute the requests in one batch.
 + Default value: `256`
 + Minimum value: greater than `0`
++ Maximum value: `10240`
 
 ### `apply-pool-size`
 
-+ The allowable number of threads in the pool that flushes data to storage. When you modify the size of this thread pool, refer to [Performance tuning for TiKV thread pools](/tune-tikv-thread-performance.md#performance-tuning-for-tikv-thread-pools).
++ The allowable number of threads in the pool that flushes data to storage, which is the size of the Apply thread pool. When you modify the size of this thread pool, refer to [Performance tuning for TiKV thread pools](/tune-tikv-thread-performance.md#performance-tuning-for-tikv-thread-pools).
```
Comment (en-jin19) [inline tikv-configuration-file.md:738]: 该修改与 #7964 中的修改重复（修改内容相同）

File: dynamic-config.md [hunk 4]
```diff
@@ -155,8 +156,10 @@ The following TiKV configuration items can be modified online:
 | `raftstore.cleanup-import-sst-interval` | The time interval to check expired SST files |
 | `raftstore.local-read-batch-size` | The maximum number of read requests processed in one batch |
 | `raftstore.hibernate-timeout` | The shortest wait duration before entering hibernation upon start. Within this duration, TiKV does not hibernate (not released). |
-| `raftstore.apply-pool-size` | The number of apply thread pool size |
-| `raftstore.store-pool-size` | The number of store thread pool size |
+| `raftstore.apply-pool-size` | The number of threads in the pool that flushes data to storage, which is the size of the Apply thread pool |
+| `raftstore.store-pool-size` | The number of threads in the pool that processes Raft, which is the size of the Raftstore thread pool |
+| `raftstore.apply-max-batch-size` | Raft state machines execute data write requests in batches by the BatchSystem. This configuration item specifies the maximum number of Raft state machines that can execute the requests in one batch. |
```
Comment (TomShawn) [inline dynamic-config.md:161]:
```suggestion
| `raftstore.apply-max-batch-size` | Raft state machines process data write requests in batches by the BatchSystem. This configuration item specifies the maximum number of Raft state machines that can execute the requests in one batch. |
```

File: dynamic-config.md [hunk 5]
```diff
@@ -155,8 +156,10 @@ The following TiKV configuration items can be modified online:
 | `raftstore.cleanup-import-sst-interval` | The time interval to check expired SST files |
 | `raftstore.local-read-batch-size` | The maximum number of read requests processed in one batch |
 | `raftstore.hibernate-timeout` | The shortest wait duration before entering hibernation upon start. Within this duration, TiKV does not hibernate (not released). |
-| `raftstore.apply-pool-size` | The number of apply thread pool size |
-| `raftstore.store-pool-size` | The number of store thread pool size |
+| `raftstore.apply-pool-size` | The number of threads in the pool that flushes data to storage, which is the size of the Apply thread pool |
```
Comment (TomShawn) [inline dynamic-config.md:159]:
```suggestion
| `raftstore.apply-pool-size` | The number of threads in the pool that spills data to disk, which is the size of the Apply thread pool |
```

File: dynamic-config.md [hunk 6]
```diff
@@ -155,8 +156,10 @@ The following TiKV configuration items can be modified online:
...
+| `raftstore.store-pool-size` | The number of threads in the pool that processes Raft, which is the size of the Raftstore thread pool |
+| `raftstore.apply-max-batch-size` | Raft state machines execute data write requests in batches by the BatchSystem. This configuration item specifies the maximum number of Raft state machines that can execute the requests in one batch. |
+| `raftstore.store-max-batch-size` | Raft state machines execute requests for log flushing in batches by the BatchSystem. This configuration item specifies the maximum number of Raft state machines that can execute the requests in one batch. |
```
Comment (TomShawn) [inline dynamic-config.md:162]:
```suggestion
| `raftstore.store-max-batch-size` | Raft state machines process requests for spilling logs into disk in batches by the BatchSystem. This configuration item specifies the maximum number of Raft state machines that can process the requests in one batch. |
```

File: tikv-configuration-file.md [hunk 7]
```diff
@@ -727,25 +728,27 @@ Configuration items related to Raftstore.
 
 ### `apply-max-batch-size`
 
-+ The maximum number of requests for data flushing in one batch
++ Raft state machines execute data write requests in batches by the BatchSystem. This configuration item specifies the maximum number of Raft state machines that can execute the requests in one batch.
```
Comment (TomShawn) [inline tikv-configuration-file.md:731]:
```suggestion
+ Raft state machines process data write requests in batches by the BatchSystem. This configuration item specifies the maximum number of Raft state machines that can execute the requests in one batch.
```

File: tikv-configuration-file.md [hunk 8]
```diff
@@ -727,25 +728,27 @@ Configuration items related to Raftstore.
...
 
-+ The allowable number of threads in the pool that flushes data to storage. When you modify the size of this thread pool, refer to [Performance tuning for TiKV thread pools](/tune-tikv-thread-performance.md#performance-tuning-for-tikv-thread-pools).
++ The allowable number of threads in the pool that flushes data to storage, which is the size of the Apply thread pool. When you modify the size of this thread pool, refer to [Performance tuning for TiKV thread pools](/tune-tikv-thread-performance.md#performance-tuning-for-tikv-thread-pools).
 + Default value: `2`
 + Minimum value: greater than `0`
 
 ### `store-max-batch-size`
 
-+ The maximum number of requests processed in one batch
++ Raft state machines execute requests for log flushing in batches by the BatchSystem. This configuration item specifies the maximum number of Raft state machines that can execute the requests in one batch.
```
Comment (TomShawn) [inline tikv-configuration-file.md:744]:
```suggestion
+ Raft state machines process requests for spilling logs into disk in batches by the BatchSystem. This configuration item specifies the maximum number of Raft state machines that can process the requests in one batch.
```

File: dynamic-config.md (hunk 5 above)
Comment (en-jin19) [reply dynamic-config.md:159]: I have checked with the author that `落盘` is more commonly translated as `flush`. 

How about changing the `flush` to `flush XXX to the disk` and updating the `落盘` to `把 XXX 落盘至磁盘 [#8784](https://github.com/pingcap/docs-cn/pull/8784)

File: tikv-configuration-file.md [hunk 9]
```diff
@@ -349,7 +349,7 @@ Configuration items related to storage.
 
 ### `scheduler-worker-pool-size`
 
-+ The number of `scheduler` threads, mainly used for checking transaction consistency before data writing. If the number of CPU cores is greater than or equal to `16`, the default value is `8`; otherwise, the default value is `4`. When you modify the size of the Scheduler thread pool, refer to [Performance tuning for TiKV thread pools](/tune-tikv-thread-performance.md#performance-tuning-for-tikv-thread-pools).
++ The number of threads in the Scheduler thread pool. Scheduler threads are mainly used for checking transaction consistency before data writing. If the number of CPU cores is greater than or equal to `16`, the default value is `8`; otherwise, the default value is `4`. When you modify the size of the Scheduler thread pool, refer to [Performance tuning for TiKV thread pools](/tune-tikv-thread-performance.md#performance-tuning-for-tikv-thread-pools).
 + Default value: `4`
 + Minimum value: `1`
```
Comment (en-jin19) [inline tikv-configuration-file.md:354]:
```suggestion
+ Value range: `[1, MAX(4, CPU)]`. In `MAX(4, CPU)`, `CPU` means the number of your CPU cores. `MAX(4, CPU)` takes the greater value out of `4` and the `CPU`.
+ Default value: MAX(4, CPU * 0.8
```

Comment (en-jin19) [reply tikv-configuration-file.md:354]: This part is revised according to [#8784](https://github.com/pingcap/docs-cn/pull/8784)

File: dynamic-config.md [hunk 10]
```diff
@@ -131,6 +131,7 @@ The following TiKV configuration items can be modified online:
 | `raftstore.raft-log-gc-threshold` | The soft limit on the maximum allowable number of residual Raft logs |
 | `raftstore.raft-log-gc-count-limit` | The hard limit on the allowable number of residual Raft logs |
 | `raftstore.raft-log-gc-size-limit` | The hard limit on the allowable size of residual Raft logs |
+| `raftstore.raft-max-size-per-msg` | The soft limit on the size of a single message packet that is allowed to be generated |
```
Comment (en-jin19) [inline dynamic-config.md:134]: This parts was updated in [#8644](https://github.com/pingcap/docs-cn/pull/8644)

File: dynamic-config.md (hunk 6 above)
Comment (en-jin19) [reply dynamic-config.md:162]: This parts was updated in [#8644](https://github.com/pingcap/docs-cn/pull/8644) and revised in [#8784](https://github.com/pingcap/docs-cn/pull/8784)

File: dynamic-config.md [hunk 11]
```diff
@@ -196,6 +199,7 @@ The following TiKV configuration items can be modified online:
 | `{db-name}.{cf-name}.hard-pending-compaction-bytes-limit` | The hard limit on the pending compaction bytes |
 | `{db-name}.{cf-name}.titan.blob-run-mode` | The mode of processing blob files |
 | `storage.block-cache.capacity` | The size of shared block cache (supported since v4.0.3) |
+| `storage.scheduler-worker-pool-size` | The number of threads in the Scheduler thread pool |
```
Comment (en-jin19) [inline dynamic-config.md:202]: This part was updated in [#8644](https://github.com/pingcap/docs-cn/pull/8644).

File: tikv-configuration-file.md [hunk 12]
```diff
@@ -349,9 +349,10 @@ Configuration items related to storage.
 
 ### `scheduler-worker-pool-size`
```
Comment (en-jin19) [inline tikv-configuration-file.md:350]: This part was updated in [#8644](https://github.com/pingcap/docs-cn/pull/8644) and revised in [#8784](https://github.com/pingcap/docs-cn/pull/8784).

File: tikv-configuration-file.md [hunk 13]
```diff
@@ -349,9 +349,10 @@ Configuration items related to storage.
...
 
-+ The number of `scheduler` threads, mainly used for checking transaction consistency before data writing. If the number of CPU cores is greater than or equal to `16`, the default value is `8`; otherwise, the default value is `4`. When you modify the size of the Scheduler thread pool, refer to [Performance tuning for TiKV thread pools](/tune-tikv-thread-performance.md#performance-tuning-for-tikv-thread-pools).
++ The number of threads in the Scheduler thread pool. Scheduler threads are mainly used for checking transaction consistency before data writing. If the number of CPU cores is greater than or equal to `16`, the default value is `8`; otherwise, the default value is `4`. When you modify the size of the Scheduler thread pool, refer to [Performance tuning for TiKV thread pools](/tune-tikv-thread-performance.md#performance-tuning-for-tikv-thread-pools).
 + Default value: `4`
-+ Minimum value: `1`
++ Value range: `[1, MAX(4, CPU)]`. In `MAX(4, CPU)`, `CPU` means the number of your CPU cores. `MAX(4, CPU)` takes the greater value out of `4` and the `CPU`.
++ Default value: MAX(4, CPU * 0.8)
```
Comment (en-jin19) [inline tikv-configuration-file.md:355]:
```suggestion
```

File: tikv-configuration-file.md [hunk 14]
```diff
@@ -491,14 +492,16 @@ Configuration items related to Raftstore.
 
 + The soft limit on the size of a single message packet
 + Default value: `"1MB"`
```
Comment (en-jin19) [inline tikv-configuration-file.md:494]: This part was updated in [#8644](https://github.com/pingcap/docs-cn/pull/8644).

File: tikv-configuration-file.md [hunk 15]
```diff
@@ -491,14 +492,16 @@ Configuration items related to Raftstore.
...
++ Minimum value: greater than `0`
++ Maximum value: `3GB`
++ Unit: KB|MB|GB
 
 ### `raft-max-inflight-msgs`
 
 + The number of Raft logs to be confirmed. If this number is exceeded, the Raft state machine slows down log sending.
 + Default value: `256`
 + Minimum value: greater than `0`
++ Maximum value: `16384`
```
Comment (en-jin19) [inline tikv-configuration-file.md:504]: This part was updated in [#8784](https://github.com/pingcap/docs-cn/pull/8784)

File: tikv-configuration-file.md [hunk 16]
```diff
@@ -728,27 +731,29 @@ Configuration items related to Raftstore.
 
 ### `apply-max-batch-size`
```
Comment (en-jin19) [inline tikv-configuration-file.md:732]: This part was updated in [#8644](https://github.com/pingcap/docs-cn/pull/8644).

File: tikv-configuration-file.md [hunk 17]
```diff
@@ -728,27 +731,29 @@ Configuration items related to Raftstore.
...
 
-+ The maximum number of requests for data flushing in one batch
++ Raft state machines process data write requests in batches by the BatchSystem. This configuration item specifies the maximum number of Raft state machines that can execute the requests in one batch.
 + Default value: `256`
 + Minimum value: greater than `0`
++ Maximum value: `10240`
 
 ### `apply-pool-size`
 
 + The allowable number of threads in the pool that flushes data to storage, which is the size of the Apply thread pool. When you modify the size of this thread pool, refer to [Performance tuning for TiKV thread pools](/tune-tikv-thread-performance.md#performance-tuning-for-tikv-thread-pools).
```
Comment (en-jin19) [inline tikv-configuration-file.md:741]:
```suggestion
+ The allowable number of threads in the pool that flushes data to the disk, which is the size of the Apply thread pool. When you modify the size of this thread pool, refer to [Performance tuning for TiKV thread pools](/tune-tikv-thread-performance.md#performance-tuning-for-tikv-thread-pools).
```

File: tikv-configuration-file.md [hunk 18]
```diff
@@ -728,27 +731,29 @@ Configuration items related to Raftstore.
 
 ### `apply-max-batch-size`
 
-+ The maximum number of requests for data flushing in one batch
++ Raft state machines process data write requests in batches by the BatchSystem. This configuration item specifies the maximum number of Raft state machines that can execute the requests in one batch.
 + Default value: `256`
 + Minimum value: greater than `0`
++ Maximum value: `10240`
 
 ### `apply-pool-size`
```
Comment (en-jin19) [inline tikv-configuration-file.md:739]: This part was updated in [#8784](https://github.com/pingcap/docs-cn/pull/8784)

File: tikv-configuration-file.md [hunk 19]
```diff
@@ -728,27 +731,29 @@ Configuration items related to Raftstore.
...
 
 + The allowable number of threads in the pool that flushes data to storage, which is the size of the Apply thread pool. When you modify the size of this thread pool, refer to [Performance tuning for TiKV thread pools](/tune-tikv-thread-performance.md#performance-tuning-for-tikv-thread-pools).
 + Default value: `2`
-+ Minimum value: greater than `0`
++ Value ranges: `[1, CPU * 10]` (`CPU` means the number of your CPU cores)
 
 ### `store-max-batch-size`
```
Comment (en-jin19) [inline tikv-configuration-file.md:745]: This part was updated in [#8644](https://github.com/pingcap/docs-cn/pull/8644) and revised in [#8784](https://github.com/pingcap/docs-cn/pull/8784)

File: tikv-configuration-file.md [hunk 20]
```diff
@@ -728,27 +731,29 @@ Configuration items related to Raftstore.
...
 
-+ The maximum number of requests processed in one batch
++ Raft state machines execute requests for log flushing in batches by the BatchSystem. This configuration item specifies the maximum number of Raft state machines that can execute the requests in one batch.
 + If `hibernate-regions` is enabled, the default value is `256`. If `hibernate-regions` is disabled, the default value is `1024`.
 + Minimum value: greater than `0`
++ Maximum value: `10240`
 
 ### `store-pool-size`
```
Comment (en-jin19) [inline tikv-configuration-file.md:752]: This part was updated in [#8784](https://github.com/pingcap/docs-cn/pull/8784)

**Comment by en-jin19** [comment]
@TomShawn @glorv Some parts have been updated according to a newly created bug-fix PR [#8784](https://github.com/pingcap/docs-cn/pull/8784). I listed PR numbers corresponding to each modification in the comments. PTAL, thanks!

**Review by glorv — COMMENTED** [review]
LGTM

File: dynamic-config.md (hunk 5 above)
Comment (TomShawn) [reply dynamic-config.md:159]: Thanks for reminding me! Changed comment to "flush data to disk".

File: dynamic-config.md (hunk 6 above)
Comment (TomShawn) [reply dynamic-config.md:162]: comment modified

File: tikv-configuration-file.md [hunk 21]
```diff
@@ -728,27 +730,29 @@ Configuration items related to Raftstore.
 
 ### `apply-max-batch-size`
 
-+ The maximum number of requests for data flushing in one batch
++ Raft state machines process data write requests in batches by the BatchSystem. This configuration item specifies the maximum number of Raft state machines that can execute the requests in one batch.
```
Comment (TomShawn) [inline tikv-configuration-file.md:733]:
```suggestion
+ Raft state machines process data write requests in batches by the BatchSystem. This configuration item specifies the maximum number of Raft state machines that can process the requests in one batch.
```

File: tikv-configuration-file.md [hunk 22]
```diff
@@ -728,27 +730,29 @@ Configuration items related to Raftstore.
...
 + Minimum value: greater than `0`
++ Maximum value: `10240`
 
 ### `apply-pool-size`
 
-+ The allowable number of threads in the pool that flushes data to storage, which is the size of the Apply thread pool. When you modify the size of this thread pool, refer to [Performance tuning for TiKV thread pools](/tune-tikv-thread-performance.md#performance-tuning-for-tikv-thread-pools).
++ The allowable number of threads in the pool that flushes data to the disk, which is the size of the Apply thread pool. When you modify the size of this thread pool, refer to [Performance tuning for TiKV thread pools](/tune-tikv-thread-performance.md#performance-tuning-for-tikv-thread-pools).
 + Default value: `2`
-+ Minimum value: greater than `0`
++ Value ranges: `[1, CPU * 10]` (`CPU` means the number of your CPU cores)
```
Comment (TomShawn) [inline tikv-configuration-file.md:742]:
```suggestion
+ Value ranges: `[1, CPU * 10]`. `CPU` means the number of your CPU cores.
```

File: tikv-configuration-file.md [hunk 23]
```diff
@@ -728,27 +730,29 @@ Configuration items related to Raftstore.
...
 + Minimum value: greater than `0`
++ Maximum value: `10240`
 
 ### `store-pool-size`
 
-+ The allowable number of threads that process Raft, which is the size of the Raftstore thread pool. When you modify the size of this thread pool, refer to [Performance tuning for TiKV thread pools](/tune-tikv-thread-performance.md#performance-tuning-for-tikv-thread-pools).
++ The allowable number of threads in the pool that processes Raft, which is the size of the Raftstore thread pool. When you modify the size of this thread pool, refer to [Performance tuning for TiKV thread pools](/tune-tikv-thread-performance.md#performance-tuning-for-tikv-thread-pools).
 + Default value: `2`
-+ Minimum value: greater than `0`
++ Value ranges: [1, CPU * 10] (`CPU` means the number of your CPU cores)
```
Comment (TomShawn) [inline tikv-configuration-file.md:755]:
```suggestion
+ Value ranges: `[1, CPU * 10]`. `CPU` means the number of your CPU cores.
```

**Pull request merged**